clio-server/src/main/scala/org/broadinstitute/clio/server/dataaccess/HttpElasticsearchDAO.scala

```diff
@@ -120,25 +125,34 @@ class HttpElasticsearchDAO private[dataaccess] (
     index: ElasticsearchIndex[_]
   ): Future[Unit] = {
     val putMappingDefinition =
-      putMapping(index.indexName / index.indexType) dynamic DynamicMapping.False as (index.fields: _*)
-    httpClient.execute(putMappingDefinition) map { response =>
-      if (!response.acknowledged)
-        throw new RuntimeException(s"""|Bad response:
-                                       |$putMappingDefinition
-                                       |$response""".stripMargin)
+      putMapping(index.indexName / index.indexType)
+        .dynamic(DynamicMapping.False)
+        .as(index.fields: _*)
+    httpClient.execute(putMappingDefinition).map { response =>
+      if (!ElasticsearchUtil.unpackResponse(response).acknowledged) {
+        throw new RuntimeException(
+          s"""|Bad response:
+              |$putMappingDefinition
+              |$response""".stripMargin
+        )
+      }
       ()
     }
   }
 
   private[dataaccess] def bulkUpdate(
     definitions: BulkCompatibleDefinition*
   ): Future[Unit] = {
-    val bulkDefinition = bulk(definitions) refresh RefreshPolicy.WAIT_UNTIL
-    httpClient.execute(bulkDefinition) map { response =>
-      if (response.errors || response.hasFailures)
-        throw new RuntimeException(s"""|Bad response:
+    val bulkDefinition = bulk(definitions).refresh(RefreshPolicy.WAIT_UNTIL)
+    httpClient.execute(bulkDefinition).map { response =>
+      val unpacked = ElasticsearchUtil.unpackResponse(response)
+      if (unpacked.errors || unpacked.hasFailures) {
```

Comment:
Why duplicate the code pattern `execute(...).map(ElasticsearchUtil.unpackResponse)`? Can we add an API layer that to avoid this code duplication, here and in our test code?